crates/ruff_python_formatter/src/expression/expr_constant.rs

```diff
@@ -88,6 +89,45 @@ impl NeedsParentheses for ExprConstant {
         } else if self.value.is_implicit_concatenated() {
             OptionalParentheses::Multiline
         } else {
+            if let AnyNodeRef::StmtAssert(
+                assert @ StmtAssert {
+                    test,
+                    msg: Some(msg),
+                    ..
+                },
+            ) = parent
+            {
+                // For asserts statements where the constant is the message and the test starts with a parenthesized node
+                //
+                // ```python
+                // assert [
+                //      a,
+                //      b,
+                //      c,
+                // ] == expected, (
+                //      "Message"
+                //  )
+                // ```
+                // split in the following order:
+                //
+                // 1. By the opening parentheses of the test
+                // 2. Add parentheses around the message
+                // 3. Add parentheses around the test
+                //
```

Comment:
Argh, this is more involved. It probably requires using `best_fitting` on the assertion side as well because we need to run through 4 variations:

* Format test, if_breaks -> don't use best fitting
* Format right, if_breaks -> don't use best fitting
* Variant 1: [test, message]
* Variant 2: [test, expand(message)],
* Variant 3: [expand(test), expand(message)],
* Variant 4: [test, message]